File: src/main/java/theoni/vkbot/Messages.java
package theoni.vkbot;

import java.io.File;

import lombok.Getter;
import theoni.vkbot.managers.ConfigManager;

public enum Messages {

    BOT_STARTED("Бот успешно запущен."),
    START(),
    USER_ID(),
    RCON(),
    RCON_WITH_COMMANDS(),
    USER_NOT_WHITELISTED(),
    COMMAND_BLOCKED(),
    FAILED_TO_CONNECT(),
    FAILED_TO_AUTHENTICATE(),
    COMMAND_SENDED(),
    RESPONSE_NULL();

    @Getter private String text;

    Messages(String text) {
        this.text = text;
    }

    Messages() {
        this.text = new ConfigManager(new File("messages.yml")).getString(name());
    }

}

File: src/main/kotlin/com/mefrreex/vkbot/config/defaults/Messages.kt
package com.mefrreex.vkbot.config.defaults

import com.mefrreex.vkbot.config.Config

enum class Messages(private val key: String) {

    USER_ID("user_id"),
    USER_NOT_WHITELISTED("user_not_whitelisted"),

    START("start"),

    RCON("rcon"),
    RCON_WITH_COMMANDS("rcon_with_commands"),

    COMMAND_BLOCKED("command_blocked"),
    FAILED_TO_CONNECT("failed_to_connect"),
    FAILED_TO_AUTHENTICATE("failed_to_authenticate"),

    COMMAND_SENDED("command_sended"),
    RESPONSE_NULL("response_null");

    companion object {

        private val CONFIG = Config("messages.yml")

        fun getConfig(): Config {
            return CONFIG
        }
    }

    fun get(): String {
        return CONFIG.getString(key)!!
    }
}